blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DatastaxExcessEnumIO.java

```diff
@@ -0,0 +1,128 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.rackspacecloud.blueflood.io.datastax;
+
+import com.codahale.metrics.Timer;
+import com.datastax.driver.core.*;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.querybuilder.Insert;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.datastax.driver.core.querybuilder.Select;
+import com.rackspacecloud.blueflood.io.CassandraModel;
+import com.rackspacecloud.blueflood.io.ExcessEnumIO;
+import com.rackspacecloud.blueflood.io.Instrumentation;
+import com.rackspacecloud.blueflood.types.Locator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.datastax.driver.core.querybuilder.QueryBuilder.bindMarker;
+
+/**
+ * This class uses the Datastax driver to read/write excess enum metrics from
+ * Cassandra metrics_excess_enums Column Family.
+ */
+public class DatastaxExcessEnumIO implements ExcessEnumIO {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DatastaxExcessEnumIO.class);
+
+    private static final String KEY = "key";
+    private static final String COLUMN1 = "column1";
+    private static final String VALUE = "value";
+
+    private PreparedStatement getValue;
+    private PreparedStatement putValue;
+
+    public DatastaxExcessEnumIO() {
+        createPreparedStatements();
+    }
+
+    /**
+     * Create all prepared statements use in this class for metrics_excess_enums
+     */
+    private void createPreparedStatements()  {
+
+        // create a generic select statement for retrieving from metrics_locator
+        Select.Where select = QueryBuilder
+                .select()
+                .all()
+                .from(CassandraModel.CF_METRICS_EXCESS_ENUMS_NAME)
+                .where();
+        getValue = DatastaxIO.getSession().prepare( select );
+
+        // create a generic insert statement for inserting into metrics_locator
+        Insert insert = QueryBuilder.insertInto( CassandraModel.CF_METRICS_EXCESS_ENUMS_NAME)
+                .value(KEY, bindMarker())
+                .value(COLUMN1, bindMarker())
+                .value(VALUE, bindMarker());
+        putValue = DatastaxIO.getSession()
+                .prepare(insert)
+                .setConsistencyLevel( ConsistencyLevel.ONE);  // TODO: remove later; required by the cassandra-maven-plugin 2.0.0-1
+                                                              // (see https://issues.apache.org/jira/browse/CASSANDRA-6238)
+
+    }
+
+    @Override
+    public Set<Locator> getExcessEnumMetrics() throws IOException {
+        Timer.Context ctx = Instrumentation.getBatchReadTimerContext(CassandraModel.CF_METRICS_EXCESS_ENUMS_NAME);
```

Comment:
@shintasmith I don't undertand.  What's BatchReadTimer?  There's no such method in Instrumentation or Timer class.